packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -966,6 +992,21 @@ export default function parse(source: string, state: State): TemplateParseResult
         }
     }
 
+    function getRenderMode(element: IRElement): LWCDirectiveRenderMode {
```

Comment:
With this PR, retrieving the render mode requires climbing up the AST. This kind of information should be present in the future `Parser` object (more details in the description).